src/Reflection/ReflectionClass.php

```diff
@@ -475,16 +475,13 @@ public function getConstructor() : ReflectionMethod
     }
 
     /**
-     * Get the properties for this class.
+     * Get only the properties for this specific class (i.e. do not search
+     * up parent classes etc.)
      *
      * @return ReflectionProperty[]
      */
-    public function getProperties() : array
+    public function getImmediateProperties() : array
     {
-        if (null !== $this->cachedProperties) {
```

Comment:
Doesn't it make more sense to cache immediate properties, so they're cached at each level of the inheritance tree, rather than caching the whole tree at each level?